guides/plugins/plugins/api/custom-pricing.md

```diff
@@ -0,0 +1,97 @@
+# Custom Pricing
+
+The Custom Pricing feature allows massive advances in the pricing model capabilities in the Shopware 6 ecosystem.
+
+The API interface exposed by this module allows the user to operate a set of commands which will enable the granular 
+''overriding'' of prices via an external data repository or ERP system. This is achieved by defining a custom 
+relationship between the current price and one of the following entities:
+- Customer
+
+### Pre-requisites and setup
+
+As Custom Pricing is part of the Commercial plugin, it requires an existing Shopware 6 installation and the activated
+Shopware 6 Commercial plugin on top. This plugin Commercial plugin can be installed as per the familiar
+[install instructions](guides/plugins/plugins/plugin-base-guide#install-your-plugin). In addition, the `Custom Prices` feature needs 
+to be activated within the relevant merchant account.
+
+## Working with the API route
+
+To create, alter and/or delete customer-specific prices, you can use the API endpoint `/api/_action/product-customer-price`. As like with any other admin request in Shopware, you first need to authenticate yourself. Therefore, please head over to the 
+[authentication guide](https://shopware.stoplight.io/docs/admin-api/ZG9jOjEwODA3NjQx-authentication) for details.
+
+Otherwise, the Custom Pricing API interface models itself upon the interface of the sync API, so you will 
+be able to package your requests similarly, see our [API documentation](https://shopware.stoplight.io/docs/admin-api).
+
+{% hint style="info" %}
+You can use the route with single `upsert` and `delete` actions or even combine those in a single request: you can pack several different commands inside one sync API request, and each of them is executed in an independent and isolated way
+{% endhint %}
+
+So, it's not surprising that the request body looks like a familiar sync request. In the payload for the `upsert` action, you pass the following data:
+* `productId`: The product whose price should be overwritten
+* `customerId`: The customer for whom we will assign a custom price
+* `price`: The new custom price you want to use.
+
+This way, we come to use a payload as seen in the example below:
+```json
+[
+  {
+    "action": "upsert",
+    "payload": [
+      {
+        "productId": "0001e32041ac451386bf9b7351c540f3",
+        "customerId": "02a3c82b5ca842c492f8656029b2e63e",
+        "price": [
+          {
+            "quantityStart": 1,
+            "quantityEnd": 10,
```

Comment:
This value makes the prices look weird, use `null` instead

Reply:
@elkmod - can this be a null ? as it must be of type integer.

Reply:
This is already outdated, please see new code

Reply:
@sushmangupta I recently updated the spotlight and Commercial plugin API definition, `null` is accepted
https://github.com/shopware/admin-api-reference/pull/2/files#diff-af2ce1e09fc283bbb187fdacd0e82764557cbe3e8141e5e4d79beda4a080656aR11762